src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.RenameFileEditor.cs

```diff
@@ -17,21 +18,28 @@ public RenameFileEditor(TService service, State state, string fileName, Cancella
             }
 
             internal override Task<ImmutableArray<CodeActionOperation>> GetOperationsAsync()
-                => Task.FromResult(RenameFileToMatchTypeName());
+                => RenameFileToMatchTypeNameAsync();
 
             /// <summary>
             /// Renames the file to match the type contained in it.
             /// </summary>
-            private ImmutableArray<CodeActionOperation> RenameFileToMatchTypeName()
+            private async Task<ImmutableArray<CodeActionOperation>> RenameFileToMatchTypeNameAsync()
             {
-                var oldDocument = SemanticDocument.Document;
-                var newDocumentId = DocumentId.CreateNewId(oldDocument.Project.Id, FileName);
+                // What I actually want to do:
+                //var documentId = SemanticDocument.Document.Id;
+                //var oldSolution = SemanticDocument.Document.Project.Solution;
+                //var newSolution = oldSolution.WithDocumentName(documentId, FileName);
 
-                return ImmutableArray.Create<CodeActionOperation>(
-                    new RenameDocumentOperation(
-                        oldDocument.Id, newDocumentId,
-                        FileName, SemanticDocument.Text),
-                    new OpenDocumentOperation(newDocumentId, activateIfAlreadyOpen: true));
+                //return ImmutableArray.Create<CodeActionOperation>(
+                //    new ApplyChangesOperation(newSolution));
+
+                // Just testing things out
+                var solution = SemanticDocument.Document.Project.Solution;
+                var symbol = State.SemanticDocument.SemanticModel.GetDeclaredSymbol(State.TypeNode, CancellationToken);
+                var documentOptions = await SemanticDocument.Document.GetOptionsAsync(CancellationToken).ConfigureAwait(false);
+                var newSolution = await Renamer.RenameSymbolAsync(solution, symbol, "WOW", documentOptions, CancellationToken).ConfigureAwait(false);
```

Comment:
??? :)

Reply:
Trying to illustrate that I can only get a global undo transaction to show up AT ALL if there is some kind of text change (which Rename File to Match Type doesn't have). With this ridiculous code, at least a global transaction shows up. Without it, I can't get anything to show on the undo stack, even with the changes to CodeActionEditHandlerService.cs which clearly go through an attempt to make one at https://github.com/dotnet/roslyn/pull/34291/files#diff-4cfc9a338393f212fa10f81440eb76b3R161